doc/meson.build

```diff
@@ -36,3 +36,22 @@ if get_option('manpages')
     install_dir: [join_paths(get_option('mandir'), 'man1'), join_paths(get_option('mandir'), 'man5')],
   )
 endif
+
+if get_option('doxygen')
```

Comment:
If sphinx-build is not found, this file exits early and never reaches the doxygen checks.

Reply:
True, this is because of the `subdir_done()` kludge earlier in this file; it's there to simplify the rules, but with Doxygen, the file needs to grow more complex.

Reply:
Actually it can be made even less complex. :)

Since 0.49.0 you can do:
```meson
find_program('fooprog', required: false, disabler: true)
```

And if it is not found, instead of doing .found() checks on it, you can simply add custom targets making use of it directly, and those targets -- because they use a disabler object in their definition -- get converted to disablers and have no effect.

So any Meson function call that attempts to use a disabler gets "infected" by the disabler, becoming a no-op. If doxygen isn't found, the attempt to build doxygen docs is no-oped and disappears. If sphinx-build isn't found, the attempt to run sphinx-build to create the manual / manpages is no-oped and disappears.

Reply:
Good hint. MPD is at `>=0.56.0`, so we can easily use this Meson feature.